upload/system/helper/Twispay_TW_Logger.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Twispay Helpers
+ *
+ * Logs messages and transactions.
+ *
+ * @author   Twistpay
+ * @version  1.0.0
+ */
+
+/* Security class check */
+if (! class_exists('Twispay_TW_Logger')) :
+    /**
+     * Class that implements methods to log
+     * messages and transactions.
+     */
+    class Twispay_TW_Logger
+    {
+        /**
+         * Function that logs a transaction to the DB.
+         *
+         * @param data Array containing the transaction data.
+         * @param that: Controller instance use for accessing runtime values like configuration, active language, etc.
+         *
+         * @return void
+         */
+        public static function twispay_tw_logTransaction($data, $that)
+        {
+            $that->load->model('extension/payment/twispay');
+            /* Extract the order. */
+            $order = $that->model_checkout_order->getOrder($data['id_cart']);
+            if ($that->model_extension_payment_twispay->checktransactions($data['transactionId'])) {
+                /* Update the DB with the transaction data. */
+                $that->db->query("UPDATE " . DB_PREFIX . "twispay_transactions SET status = '" . $data['status'] . "' WHERE transactionId = '%d'", $data['transactionId']);
+            } else {
+                /* Insert transaction data into DB. */
+                $that->db->query("INSERT INTO `" . DB_PREFIX . "twispay_transactions` (`status`, `id_cart`, `identifier`, `orderId`, `transactionId`, `customerId`, `cardId`) VALUES ('" . $data['status'] . "', '" . $data['id_cart'] . "', '" . $data['identifier'] . "', '" . $data['orderId'] . "', '" . $data['transactionId'] . "', '" . $data['customerId'] . "', '" . $data['cardId'] . "');");
```

Comment:
All data used inside DB query must be properly escaped (in both queries present here).

Reply:
Will close because was fixed in last commit.